### PR TITLE
Refine TWF by adding independent mus and mud and a special twf for TPV22; Update the corresponding user manual

### PR DESCRIPTION
**Pull request by Huihuiweng**
Add a special kind of TWF for TPV benchmarks

Revise a bug in special TWF

minor changes

small bug

Update 06_fault_sources.tex in user manual

### CODE REVIEW

**Comment by Huihuiweng** [comment]
What are the causes of the failure of CI /Test on Mac? This is weird because I can successfully compile it on my computer.

**Comment by danielpeter** [comment]
will figure out the issues later, seems to be related to MPI and the parallel code compilation...

**Comment by Huihuiweng** [comment]
> will figure out the issues later, seems to be related to MPI and the parallel code compilation...

Thanks.

File: my_mpi.mod
Comment (danielpeter) [inline my_mpi.mod:1]: this file caused the test problems - it shouldn't be included as it is a pre-compiled module file that only works for specific compilers & systems. i will take it out in the next PR...

**Pull request merged**